expo/expo
# [autolinking] Link only project's dependencies

# Why

The script was currently autolinking each package it found in workspace root's node_modules, so including packages that are not the dependencies of the subproject.

# How

Traversing through project's dependencies and filtering out packages that are not project's dependencies. Projects will need to explicitly specify the dependencies in `package.json` to autolink them.

It turned out that NCL doesn't specify some packages in `dependencies` — I added them so that bare-expo autolinks the same packages as without these changes

# Test Plan

- Run `yarn run expo-modules-autolinking search -p ios` in bare-expo and notice it returns the correct list of packages
- Remove `expo-screen-orientation` dependency from NCL and re-yarn
- Run the search command again and check that this package is no longer returned (it's no longer a dependency of bare-expo)
